pallets/loans/src/math.rs

```diff
@@ -356,54 +356,60 @@ mod tests {
 
 	#[test]
 	fn test_valid_write_off_groups() {
-		let groups = vec![
+		let groups: Vec<WriteOffGroup<Rate>> = vec![
 			WriteOffGroup {
-				percentage: (),
+				percentage: Rate::saturating_from_rational(0, 100),
 				overdue_days: 3,
+				penalty_interest_rate_per_sec: Rate::saturating_from_rational(0, 100),
 			},
 			WriteOffGroup {
-				percentage: (),
+				percentage: Rate::saturating_from_rational(0, 100),
 				overdue_days: 5,
+				penalty_interest_rate_per_sec: Rate::saturating_from_rational(0, 100),
 			},
 			WriteOffGroup {
-				percentage: (),
+				percentage: Rate::saturating_from_rational(0, 100),
 				overdue_days: 6,
+				penalty_interest_rate_per_sec: Rate::saturating_from_rational(0, 100),
 			},
 			WriteOffGroup {
-				percentage: (),
+				percentage: Rate::saturating_from_rational(0, 100),
 				overdue_days: 14,
+				penalty_interest_rate_per_sec: Rate::saturating_from_rational(0, 100),
 			},
 			WriteOffGroup {
-				percentage: (),
+				percentage: Rate::saturating_from_rational(0, 100),
 				overdue_days: 9,
+				penalty_interest_rate_per_sec: Rate::saturating_from_rational(0, 100),
 			},
 			WriteOffGroup {
-				percentage: (),
+				percentage: Rate::saturating_from_rational(0, 100),
 				overdue_days: 7,
+				penalty_interest_rate_per_sec: Rate::saturating_from_rational(0, 100),
 			},
 		];
 
 		let sec_per_day = seconds_per_day();
 
 		// maturity date in days and current time offset to maturity date  and resultant index from the group
-		let tests: Vec<(Moment, Moment, Option<u32>)> = vec![
+		let tests: Vec<(Moment, Moment, Option<(u32, &WriteOffGroup<Rate>)>)> = vec![
 			// day 0, and now is at zero, index is None
 			(0, 0, None),
 			(0, 1, None),
 			// now is 3 and less than 5 days, the index is valid
-			(0, 3, Some(0)),
-			(0, 4, Some(0)),
+			(0, 3, Some((0, &groups[0]))),
+			(0, 4, Some((0, &groups[0]))),
 			// now is 5 and less than 6 days, the index is valid
-			(0, 5, Some(1)),
+			(0, 5, Some((1, &groups[1]))),
 			// now is 6 and less than 7 days, the index is valid
-			(0, 6, Some(2)),
+			(0, 6, Some((2, &groups[2]))),
 			// now is 7 and 8 and less than 9 days, the index is valid
-			(0, 7, Some(5)),
-			(0, 8, Some(5)),
+			(0, 7, Some((5, &groups[5]))),
+			(0, 8, Some((5, &groups[5]))),
```

Comment:
Is the index and the groups element always needing to be related or is that just a test convenience?

Reply:
`valid_write_off_group` returns the tuple of `(idx, &group)`, so these do need to match in the tests